integration/test_ml.py

```diff
@@ -0,0 +1,389 @@
+#'pip install tensorflow=2.0.0' for version 2 etc.
+if _TF_ENABLED:
+    # Test tensor flow conversion functions if tensor flow is enabled.
+    SAVED_MODEL_DIR = '/tmp/saved_model/1'
+
+    def _clean_up_tmp_directory():
+        if os.path.exists(SAVED_MODEL_DIR):
+            shutil.rmtree(SAVED_MODEL_DIR)
+
+    @pytest.fixture
+    def keras_model():
+        x_array = [-1, 0, 1, 2, 3, 4]
+        y_array = [-3, -1, 1, 3, 5, 7]
+        model = tf.keras.models.Sequential(
+            [tf.keras.layers.Dense(units=1, input_shape=[1])])
+        model.compile(optimizer='sgd', loss='mean_squared_error')
+        model.fit(x_array, y_array, epochs=3)
+        yield model
+
+    @pytest.fixture
+    def saved_model_dir(keras_model):
+        # different versions have different model conversion capability
+        # pick something that works for each version
+        save_dir = SAVED_MODEL_DIR
+        _clean_up_tmp_directory() # previous failures may leave files
+        if tf.version.VERSION.startswith('1.'):
+            tf.reset_default_graph()
+            x_var = tf.placeholder(tf.float32, (None, 3), name="x")
+            y_var = tf.multiply(x_var, x_var, name="y")
+            with tf.Session() as sess:
+                tf.saved_model.simple_save(sess, save_dir, {"x": x_var}, {"y": y_var})
+        else:
+            # If it's not version 1.x or version 2.x we need to update the test.
+            assert tf.version.VERSION.startswith('2.')
+            tf.saved_model.save(keras_model, save_dir)
+        yield save_dir
+
+
+    def test_from_keras_model(keras_model):
+        source1 = ml.TFLiteGCSModelSource.from_keras_model(keras_model, 'model2.tflite')
+        assert re.search(
+            '^gs://.*/Firebase/ML/Models/model2.tflite$',
+            source1.gcs_tflite_uri) is not None
+
+        # Validate the conversion by creating a model
+        format1 = ml.TFLiteFormat(model_source=source1)
+        model1 = ml.Model(display_name="KerasModel1", model_format=format1)
+        firebase_model = ml.create_model(model1)
+        assert firebase_model.model_id is not None
+        assert firebase_model.validation_error is None
+
+        _clean_up_model(firebase_model)
+
+    def test_from_saved_model(saved_model_dir):
+        # Test the conversion helper
+        source1 = ml.TFLiteGCSModelSource.from_saved_model(saved_model_dir, 'model3.tflite')
+        assert re.search(
+            '^gs://.*/Firebase/ML/Models/model3.tflite$',
+            source1.gcs_tflite_uri) is not None
+
+        # Validate the conversion by creating a model
+        format1 = ml.TFLiteFormat(model_source=source1)
+        model1 = ml.Model(display_name="SavedModel1", model_format=format1)
+        firebase_model = ml.create_model(model1)
+        assert firebase_model.model_id is not None
+        assert firebase_model.validation_error is None
+
+        _clean_up_model(firebase_model)
```

Comment:
Try-finally to delete the model. Let the fixture clean up the temp directory.

Reply:
done